FabNESO/tasks.py

```diff
@@ -252,3 +265,309 @@ def neso_ensemble(
+    efficient parameter calibration in expensive to simulate models. Here we use the
+    Python implementation of the algorithm in the package PyVBMC (Huggins et al., 2023).
+
+    Args:
+        config: Directory with ensemble configuration information.
+        solver: Which NESO solver to use.
+        conditions_file_name: Name of conditions XML file in configuration directory.
+        mesh_file_name: Name of mesh XML in configuration directory.
+        processes: Number of processes to run in each job in the ensemble.
+        nodes: Number of nodes to run on. Only applicable when running on a multi-node
+            system.
+        cpus_per_process: Number of processing units to use per process. Only
+            applicable when running on a multi-node system.
+        wall_time: Maximum time to allow job to run for. Only applicable when submitting
+            to a job scheduler.
+        reference_field_file: Name of a numpy file that holds a reference field
+            measurement for the calibration run
+        **vbmc_parameters: The parameters to be scanned in the VBMC instance. The value
+            is a colon separated list: lower bound: upper bound: plausible lower bound
+            : plausible upper bound
+
+    References:
+        1. Acerbi, L. (2018). Variational Bayesian Monte Carlo.
+           Advances in Neural Information Processing Systems, 31.
+        2. Huggins et al., (2023). PyVBMC: Efficient Bayesian inference in Python.
+           Journal of Open Source Software, 8(86), 5428,
+           https://doi.org/10.21105/joss.05428
+
+    """
+    # Create the output directory
+    output_directory = Path(output_directory_name)
+    if not output_directory.is_dir():
+        output_directory.mkdir(parents=True)
+
+    if not Path(reference_field_file).is_file():
+        msg = (
+            f"reference_field_file {reference_field_file} not found. "
+            "One must be specificed for the calibration to run. If in doubt, please "
+            "run the neso_write_field task to write a measured field to file first."
+        )
+        raise ValueError(msg)
+
+    # Contains a number of items that enable the vbmc running of NESO
+
+    # These ensure that the field information is written out, which is not done
+    # be default in the conditions files.
+    # This would need to be changed per solver
+    para_overrides = {
+        "particle_num_write_field_steps": 100,
+        "line_field_deriv_evaluations_step": 20,
+        "line_field_deriv_evaluations_numx": 100,
+        "line_field_deriv_evaluations_numy": 1,
+    }
+
+    # Put all of the NESO arguments in one dict
+    neso_args = _create_job_args_dict(
+        solver,
+        conditions_file_name,
+        mesh_file_name,
+        processes,
+        nodes,
+        cpus_per_process,
+        wall_time,
+    )
+
+    # Make this read in from a config too?
+    observation_noise_std = 0.1
```

Comment:
I think just having this passed in as an argument to the task would be the easiest way to let user change.